avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1058,12 +1058,44 @@ private Object toConnectData(Schema schema, Object value) {
             arrayVal =
             recordValue.get(ANYTHING_SCHEMA.getField(ANYTHING_SCHEMA_ARRAY_FIELD).pos());
         if (arrayVal != null) {
-          return toConnectData(SchemaBuilder.array(null).build(), arrayVal);
+          // We cannot reuse the logic like we do in other cases because it is not possible to
+          // construct an array schema with a null item schema, but we do not know the schema of the
+          // items.
+          if (!(arrayVal instanceof Collection)) {
```

Comment:
Is it really possible for this to be a`Collection`? I'm asking because in the construction it seems that a `List` is passed in the case of an `ARRAY` too. Or I'm missing a construction case. 

We could change casting here, or if we use `Collection` below as well, we could reserve for ourselves the option to use another type of `Collection` in the future (e.g. if we ever wanted to use a `Set` instead of a `List` - this example is arbitrary)